File: ePreschool.Core/Entities/Message.cs
using ePreschool.Core.Entities.Base;

namespace ePreschool.Core.Entities
{
    public class Message : BaseEntity
    {
        public string? Text { get; set; } = default!;
        public int FromUserId { get; set; }
        public Person FromUser { get; set; } = default!;
        public int ToUserId { get; set; }
        public Person ToUser { get; set; } = default!;
    }
}

File: UI/epreschool_mobile/lib/models/message.dart
import 'package:epreschool_mobile/models/person.dart';

import 'base.dart';

class Message extends BaseModel {
  late String text;
  late int fromUserId;
  late Person fromUser;
  late int toUserId;
  late Person toUser;

  Message();

  Message.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    createdAt = json['createdAt'];
    text = json['text'];
    fromUserId=json['fromUserId'];
    if (json['fromUser'] != null) {
      fromUser = Person.fromJson(json['fromUser']);
    }
    toUserId = json['toUserId'];
    if (json['toUser'] != null) {
      toUser = Person.fromJson(json['toUser']);
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['text'] = text;
    data['fromUserId'] = fromUserId;
    data['toUserId'] = toUserId;
    return data;
  }
}